src/core/service.c

```diff
@@ -1561,21 +1561,43 @@ static int service_spawn(
                 }
         }
 
-        if (flags & EXEC_SETENV_RESULT) {
-                if (asprintf(our_env + n_env++, "SERVICE_RESULT=%s", service_result_to_string(s->result)) < 0)
-                        return -ENOMEM;
+        if (flags & (EXEC_SETENV_RESULT | EXEC_SETENV_MONITOR_RESULT)) {
+                /* We don't expect both EXEC_SETENV_RESULT and EXEC_SETENV_MONITOR_RESULT to be set. */
+                assert(!((flags & EXEC_SETENV_RESULT) && (flags & EXEC_SETENV_MONITOR_RESULT)));
 
-                if (s->main_exec_status.pid > 0 &&
-                    dual_timestamp_is_set(&s->main_exec_status.exit_timestamp)) {
-                        if (asprintf(our_env + n_env++, "EXIT_CODE=%s", sigchld_code_to_string(s->main_exec_status.code)) < 0)
-                                return -ENOMEM;
+                const char *env_prefix = (flags & EXEC_SETENV_MONITOR_RESULT) ? "MONITOR_" : "";
```

Comment:
env_prefix is used only in the next if block, so it can be declared inside it

Reply:
ack.